service/rpc/servicer.go

```diff
@@ -188,6 +190,47 @@ func (s *Servicer) AccountBalance(ctx context.Context, request *types.AccountBal
 		return createResponse(NewAmount(goldAmt, CeloGold)), nil
 	}
 
+	if subAccount.Address[:len("ReleaseGold")] == "ReleaseGold" {
+		releaseGold, err := contracts.NewReleaseGold(accountAddr, s.cc.Eth)
+		if err != nil {
+			return nil, LogErrValidation(errors.New("Account address must be a ReleaseGold instance"))
+		}
+
+		beneficiary, err := releaseGold.Beneficiary(requestedBlockOpts)
+		if err != nil {
+			return nil, LogErrCeloClient("ReleaseGold.beneficiary", err)
+		}
+
+		if subAccount.Metadata != nil {
+			if addrStr, ok := subAccount.Metadata["beneficiary"]; ok {
+				providedBeneficiary := common.HexToAddress(addrStr.(string))
+				if beneficiary != providedBeneficiary {
+					return nil, LogErrValidation(errors.New("Provided beneficiary address does not match ReleaseGold contract"))
+				}
+			}
+		}
+
+		var celoGoldAmount *big.Int
+		switch subAccount.Address {
```

Comment:
It feels a bit weird to use `Address` here as it normally is of type `Common.Address`

Reply:
it feels more like an `AddressType` or so